packages/node_modules/@webex/internal-plugin-ediscovery/src/retry.js

```diff
@@ -0,0 +1,34 @@
+const retryErrors = [429, 502, 503, 504];
+
+async function requestWithRetries(ctx, func, args, retryCount = 0, retryIntervalInSeconds = 0, maxRetries = 3) {
+  await timeout(retryIntervalInSeconds);
+
+  return func.apply(ctx, args)
+    .catch((reason) => {
+      if (retryErrors.includes(reason.statusCode) && retryCount < maxRetries) {
+        retryCount += 1;
+        let retryIntervalInSeconds = (retryCount + 1) ** 2; // 4, 9 and 16 second delays as default
+
+        if (reason.headers && reason.headers['retry-after']) {
+          retryIntervalInSeconds = reason.headers['retry-after'];
+        }
+        console.error(`Request #${retryCount} error: ${reason.statusCode}. Attempting retry #${retryCount} in ${retryIntervalInSeconds} seconds`);
+
+        return requestWithRetries(ctx, func, args, retryCount, retryIntervalInSeconds, maxRetries);
+      }
+
+      return Promise.reject(reason);
+    });
+}
```

Comment:
New reusable function that takes a request and can call it multiple times if rate-limited, or the service is unresponsive.

Reply:
There is a retry in `packages/node_modules/@webex/common/src/retry.js`. Can this core one be used or modified to keep retry logic in one place?

Reply:
I don't think it can Matt, it doesn't seem to provide everything we need in terms of retrying various responses status codes and error handling. It doesn't appear to be used anywhere else in the sdk, and I couldn't find any tests to illustrate how it should be used, but I'm also reluctant to modify it. I wasn't aware this class existed, I was under the impression that each plugin had to handle retries in their own way, e.g. plugin-meetings.